src/ixdat/techniques/deconvolution.py

```diff
@@ -0,0 +1,241 @@
+        )
+        partial_current = partial_current * sum(kernel)
+        return t_sig, partial_current
+
+    def extract_kernel(self, signal_name, cutoff_pot=0, tspan=None, tspan_bg=None):
+        """Extracts a Kernel object from a measurement.
+
+        Args:
+            signal_name (str): Signal name from which the kernel/impule
+                response is to be extracted.
+            cutoff_pot (int): Potential which the defines the onset of the
+                impulse. Must be larger than the resting potential before the
+                impulse.
+            tspan(list): Timespan from which the kernel/impulse response is
+                extracted.
+            tspan_bg (list): Timespan that corresponds to the background signal.
+        """
+        x_curr, y_curr = self.grab_current(tspan=tspan)
+        x_pot, y_pot = self.grab_potential(tspan=tspan)
+        x_sig, y_sig = self.grab_signal(signal_name, tspan=tspan, tspan_bg=tspan_bg)
+
+        if signal_name == "M32":
+            t0 = x_curr[np.argmax(y_pot > cutoff_pot)]  # time of impulse
+        elif signal_name == "M2" or signal_name == "M17":
+            t0 = x_curr[np.argmax(y_pot < cutoff_pot)]
+        else:
+            print("mass not found")
+
+        x_sig = x_sig - t0
+
+        y_sig = y_sig[x_sig > 0]
+        x_sig = x_sig[x_sig > 0]
+
+        y_curr = y_curr[x_curr > t0]
+        x_curr = x_curr[x_curr > t0]
+        y_pot = y_pot[x_pot > t0]
+        x_pot = x_pot[x_pot > t0]
+
+        kernel = Kernel(
+            MS_data=np.array([x_sig, y_sig]),
+            EC_data=np.array([x_curr, y_curr, x_pot, y_pot]),
+        )
+
+        return kernel
+
+
+class Kernel:
+    """Kernel class implementing datatreatment of kernel/impulse response data."""
+
+    # TODO: Make class inherit from Measurement, add properties to store kernel
+    # TODO: Reference equations to paper.
+    def __init__(
+        self,
+        parameters={},
+        MS_data=None,
+        EC_data=None,
+    ):
+        """Initializes a Kernel object either in functional form by defining the
+        mass transport parameters or in the measured form by passing of EC-MS
+        data.
+
+        Args:
+            parameters (dict): Dictionary containing the mass transport
+                parameters with the following keys:
+                    diff_const: Diffusion constant in liquid
+                    work_dist: Working distance between electrode and gas/liq interface
+                    vol_gas: Gas sampling volume of the chip
+                    volflow_cap: Volumetric capillary flow
+                    henry_vola: Dimensionless Henry volatility
+                MS_data (list): List of numpy arrays containing the MS signal
+                    data.
+                EC_data (list): List of numpy arrays containing the EC (time,
+                    current, potential).
+        """
+
+        if MS_data is not None and parameters:  # TODO: Make two different classes
```

Comment:
if MS_data is a list if given and an empty one probably doesn't make sense, so it could be `if MS_data and parameters`